User detail view page

silviczka opened the pull request:
Create UserDetailView page with mock API request

silviczka committed:
Add UserDetailView page with mock API request

silviczka committed:
Merge branch 'dev' of https://github.com/MusicaSacraOrg/cantuscatholici-frontend into dev

SamuelSlavik committed:
Fix eslint

SamuelSlavik commented on src/views/notFoundView/NotFoundView.tsx:
```diff
@@ -7,7 +7,7 @@ export function NotFoundView() {
         <Prince isPageLayout={true} className={bem()}>
             <h2>404</h2>
-            <p>Ľutujeme, zadanú adresu sa nepodarilo náisť</p>
+            <p>Ľutujeme, zadanú adresu sa nepodarilo nájsť</p>
```
Hehe ďakujem :)

SamuelSlavik commented on scripts/generate-scss-imports.cjs:
```diff
@@ -1,9 +1,15 @@
 const glob = require('glob');
 const fs = require('fs');
+const path = require('path');
```
Potrebujeme import aj na `path`?

silviczka replied:
removed in commit > "fix: normalize line endings and resolve all linting errors"

SamuelSlavik commented on src/api/user/UserEndpoints.ts:
```diff
@@ -2,10 +2,12 @@ import { Endpoints } from '../Endpoints';
     static getCurrentUser() {
-        return `${this.baseUrl}/user/`;
+        //return `${this.baseUrl}/user/`;
+        return '/mocks/user.json';
```
Toto mockovať netreba, toto by z backendu už malo chodiť. Ale vlastne môžeme úplne odmazať, keďže rovnaký endpoint je v AuthEndpoints.ts

silviczka replied:
fixed in upcoming commit

SamuelSlavik commented on src/api/user/UserService.ts:
```diff
@@ -0,0 +1,28 @@
+            this.getHeaders()
+        );
+
+        const user = response.data.users.find((u) => u.id.toString() === id);
```
endpoint getUser z backendu bude vracať vždy len jedného usera. Viem že v mockoch to nepasuje, ale treba radšej použiť mock kde bude len jeden objekt usera a nie list userov. Potom to nemusíme transformovať. 

Listing bude neskôr implementovať factory samotná (ale to navrhneme neskôr spolu).

Tým pádom môžeme potom odmazať aj error či return a nepotrebujeme ani ukladať response do premennej. Bude sa to v podstate veľmi podobať na `getCurrentUser()` z `AuthService.ts`

silviczka replied:
fixed in upcoming commit

SamuelSlavik commented on src/api/user/useUserDetail.tsx:
```diff
@@ -0,0 +1,70 @@
+} from '@musica-sacra/notifications';
+import { Loader } from '@musica-sacra/loader';
+
+export function useUserDetail(id: string | undefined | null) {
```
Premenoval by som na useGetUser alebo useGetUserDetail. Nieje to chyba, ale bude za tým väčší koncept o ktorom si ešt povieme

Taktiež `id` by mala byť určite vždy defined. Nemôžem hľadať usera ak nemám id.

silviczka replied:
fixed in upcoming commit

SamuelSlavik commented on src/api/user/useUserDetail.tsx:
```diff
@@ -0,0 +1,70 @@
+    const query = useQuery({
+        queryKey: ['user', id],
+        queryFn: async () => {
+            if (!id) throw new Error('No id provided');
```
Keď vynútime `id`, tak nemusime kontrolovat ci sme ju dostali, typescript nas automaticky nepusti.

silviczka replied:
fixed in upcoming commit

SamuelSlavik commented on src/api/user/useUserDetail.tsx:
```diff
@@ -0,0 +1,70 @@
+        },
+        enabled: !!id,
+        staleTime: Infinity,
+        gcTime: Infinity,
```
staleTime ani gcTime nieje potrebne nastavovat na infinity. gcTime ma default hodnotu 5 minut, co pri vacsine beznych dat je uplne dostacujuce. staleTime asi taktiez nemusime zatial nastavovat, nevieme zarucit ze tieto data bude aktualne navzdy, mozu sa menit v case. 

Tieto dva flagy je useful nastavit pri implementacii prihlaseneho uzivatela, kde manualne nastavujeme validitu dat a nechceme aby ich reactQuery niekedy prenacitala automaticky. V ostatnych pripadoch to je asi naopak ziadane, aby reactQuery automaticky prenacitavala a updatovala data.

silviczka replied:
fixed in upcoming commit

SamuelSlavik commented on src/views/userView/UserDetailView.scss:
```diff
@@ -0,0 +1,99 @@
+    max-width: 800px;
+  }
+
+  &__profile,
```
Tuto je velmi dobra konvencia pisat tie nested rules takymto stylom:

```scss
.view-user {
  &__ {
    &profile {...}
    
    &songs {...}
    
    ...
  }
}
```

silviczka replied:
fixed in upcoming commit

SamuelSlavik commented on src/views/userView/UserDetailView.tsx:
```diff
@@ -0,0 +1,116 @@
+                    <div className={bem('profile')}>
+                        <div className={bem('profile-inner')}>
+                            <div className={bem('avatar')} aria-hidden>
+                                {((user as any).avatar as string) ? (
```
`avatar` mozme priamo pridat do typu User aby sme nemuseli pouzivat as any.

silviczka replied:
removed in commit > "fix: normalize line endings and resolve all linting errors"

SamuelSlavik commented on src/views/userView/UserDetailView.tsx:
```diff
@@ -0,0 +1,116 @@
+    const { addNotification } = useContext(NotificationsContext);
+
+    const description =
+        (user && (user.description || (user as any).descriptions)) || '';
```
user.descriptions urcite nebudu, vzdy to bude len jeden string, maximalne ako rich content. Takze tento riadok tu vobec nemusi byt, a priamo v jsx mozme pouzit user.description

silviczka replied:
fixed in upcoming commit

SamuelSlavik commented on src/views/userView/UserDetailView.tsx:
```diff
@@ -0,0 +1,116 @@
+                                                }
+                                            >
+                                                Skryť číslo
+                                            </button>
```
Tlacitko na hide-number nebude treba. Ono sa to skryva kvoli robotom ktory scrapuju weby. Ako ochrana nam teda staci skryt to na zaciatku, ked uz sa to odkryje, tak uz to nemusime znova skryvat.

silviczka replied:
removed in upcoming commit

SamuelSlavik commented on src/views/userView/UserDetailView.tsx:
```diff
@@ -0,0 +1,116 @@
+                                            </span>
+                                            <button
+                                                className={bem('copy-number')}
+                                                onClick={async () => {
```
Tuto funkciu by som definoval mimo jsx a tu ju uz len zavolal

silviczka replied:
fixed in upcoming commit

SamuelSlavik commented on src/views/userView/UserDetailView.tsx:
```diff
@@ -0,0 +1,116 @@
+                                                            'Tel. číslo skopírované',
+                                                            NotificationTypes.SUCCESS
+                                                        );
+                                                    } catch (e) {
```
Aby bol linter spokojny, tak vylogujme do console napriklad

silviczka replied:
fixed in upcoming commit

SamuelSlavik commented on src/views/userView/UserDetailView.tsx:
```diff
@@ -0,0 +1,116 @@
+} from '@musica-sacra/notifications';
+
+export function UserDetailView() {
+    const { bem } = useBem('view-user');
```
premenoval by som classname na `view-user-detail`, nech je to konzistentné. Taktiež aj folder kde je tento súbor

silviczka replied:
fixed in upcoming commit

SamuelSlavik commented on src/views/userView/UserDetailView.tsx:
```diff
@@ -0,0 +1,116 @@
+                                            >
+                                                {user.mobile}
+                                            </span>
+                                            <button
```
Super nápad že si rovno pridala aj tlačítko na kopírovanie, vidno že ti to myslí ako softwarovému inžinierovi. K dokonalosti by som možno ale to tlačítko vymenil za ikonku kopírovania. 

Ako dobrý zdroj sa mi osvedčilo priamo kopírovať svg ikoniek z https://fonts.google.com/icons?icon.set=Material+Icons&selected=Material+Icons+Round:content_copy:&icon.query=copy&icon.size=24&icon.color=%23e3e3e3&icon.style=Rounded
Netreba nic instalovat, takze nas bundle ostava pekne malicky.

silviczka replied:
fixed in upcoming commit

SamuelSlavik commented on src/views/userView/UserDetailView.tsx:
```diff
@@ -0,0 +1,116 @@
+                            </div>
+
+                            <div className={bem('main')}>
+                                <div className={bem('profile-header')}>
```
Z tohto by som spravil samostatnu komponentu, urcite to vieme vyuzit napriklad aj v sidebare administracie.

silviczka replied:
fixed in upcoming commit

SamuelSlavik commented on src/views/userView/UserDetailView.tsx:
```diff
@@ -0,0 +1,116 @@
+            {isError && <div>Chyba pri načítaní používateľa</div>}
+            {!isLoading && user && (
+                <>
+                    <div className={bem('profile')}>
```
Aj z tohto celého by som vlastne spravil samostatnú komponentu, zíde sa nám to aj pri detaile v admine.

silviczka replied:
fixed in upcoming commit

silviczka committed:
fix: normalize line endings and resolve all linting errors

silviczka committed:
fix: refactor user detail view based on code review feedback

silviczka commented:
commit -> fix: refactor user detail view based on code review feedback

Hello @SamuelSlavik, pls let me know if anything needs to be fixed further! 😊

**Summary:**
Refactored user detail functionality based on code review feedback, including component extraction, API service simplification, and naming consistency improvements.

**Key Changes:**

**API & Hooks:**
- Simplified UserService.getUser() to return single user object (matching AuthService.getCurrentUser() pattern)
- Renamed useUserDetail → useGetUser with required id parameter
- Removed staleTime and gcTime from useGetUser to use React Query defaults
- Updated mock endpoint to return single user object

**Component Extraction:**
- Extracted ProfileHeader component (reusable for admin views)
- Extracted UserProfile component (encapsulates full profile display logic)
- Added Storybook stories for both new components

**Naming & Structure:**
- Renamed userView folder → userDetailView for consistency
- Renamed class name view-user → view-user-detail
- Unified role casing to capitalized format ("Admin", "Redactor", "User") across mocks and types

**Data Model:**
- Fixed mock data: "descriptions" → "description" to match TypeScript model
- Removed unused descriptions field from UserDetail type

**UI Improvements:**
- Removed "Skryť číslo" button (simplified phone number visibility)
- Extracted copy function outside JSX
- Added error logging for copy functionality

**Storybook:**
- Fixed Storybook configuration for ES modules compatibility
- Added context providers (QueryClient, UserProvider, NotificationsProvider)
- Updated stories to use Roles constants for type safety

SamuelSlavik committed:
Merge branch 'dev' into user-detail-view
